test.js

```diff
@@ -49,3 +49,72 @@ test('Queue namespaced data until `out` is set', t => {
 	])
 	t.end()
 })
+
+test('Rerouting `out`', t => {
+	const log = logger('myservice')
+
+	log('level 1 A')
+
+	const sublog = log.create('downstream')
+
+	sublog('level 2 A')
+
+	const buf = []
+	log.out(e => buf.push(e))
+
+	log('level 1 B')
+
+	const buf2 = []
+	log.out(e => buf2.push(e))
+
+	log('level 1 C')
+	sublog('level 2 B')
+	log('level 1 D')
+
+	t.deepEqual(buf, [
+		['myservice', 'level 1 A'],
+		['myservice', 'downstream', 'level 2 A'],
+		['myservice', 'level 1 B'],
+	])
+	t.deepEqual(buf2, [
+		['myservice', 'level 1 C'],
+		['myservice', 'downstream', 'level 2 B'],
+		['myservice', 'level 1 D'],
+	])
+	t.end()
+})
+
+test('Queue data while `out` is detached', t => {
+	const log = logger('myservice')
+
+	log('level 1 A')
+
+	const sublog = log.create('downstream')
+
+	sublog('level 2 A')
+	sublog('level 2 B')
+
+	const buf = []
+	log.out(e => buf.push(e))
+	t.deepEqual(buf, [
+		['myservice', 'level 1 A'],
+		['myservice', 'downstream', 'level 2 A'],
+		['myservice', 'downstream', 'level 2 B'],
+	])
```

Comment:
I think you should move this to the end to prove that it stops writing to this output once it is detached.